stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -890,3 +890,54 @@ end
 *(A::Adjoint{<:Any,<:RealHermSymComplexHerm}, B::Transpose{<:Any,<:AbstractTriangular}) = A.parent * B
 *(A::Transpose{<:Any,<:RealHermSymComplexSym}, B::Adjoint{<:Any,<:AbstractTriangular}) = A.parent * B
 *(A::Transpose{<:Any,<:RealHermSymComplexSym}, B::Transpose{<:Any,<:AbstractTriangular}) = A.parent * B
+
+"""
+    symmetrize!(X::AbstractMatrix; conjugate::Bool=true)
+
+Make the matrix `X` symmetric in-place using the formula `(X + X') / 2`.
+If `conjugate` is `true`, `X` is made Hermitian rather than symmetric.
+
+See also: [`symmetrize`](@ref)
+
+!!! compat "Julia 1.3"
+    This function requires Julia 1.3 or later.
+"""
+symmetrize!(X::AbstractMatrix; conjugate::Bool=true) =
+    conjugate ? _symmetrize!(conj, real, X) : _symmetrize!(identity, identity, X)
+
+function _symmetrize!(f::Function, g::Function, X::AbstractMatrix)
+    inds = axes(X, 1)
+    inds == axes(X, 2) || throw(DimensionMismatch("matrix is not square"))
+    r = first(inds)
+    s = step(inds)
+    @inbounds for j in inds
+        X[j,j] = g(X[j,j])
+        for i in r:s:j-s
+            X[i,j] = (X[i,j] + f(X[j,i])) / 2
+            X[j,i] = f(X[i,j])
+        end
+    end
+    X
+end
+
+const ClosedUnderDivision = Union{AbstractFloat, Complex{<:AbstractFloat},
+                                  Rational, Complex{<:Rational}}
```

Comment:
This doesn't seem very general.

Why not use
```
hermitian(X::AbstractMatrix) = hermitian!(copyto!(similar, X, typeof((isempty(X) ? zero(eltype(X)) : first(X))/2)), X))
```
instead?

Reply:
If we aren't supporting the array-of-arrays case (and your `conj` code above does not support this), then we should restrict ourselves to `AbstractMatrix{T}` where `{T<:Number}` and then do `typeof(zero(T)/2))`.

Reply:
Good notes. Do we care about the array-of-arrays case? I hadn't considered it when writing this initially.

Reply:
I would just restrict to `AbstractMatrix{T} where {T<:Number}` for now.   I think it's fine not to support the array-of-arrays case at least initially, as long as we don't silently give the wrong answers for that case.

Reply:
Restricted to numbers and simplified per your suggestions.